salt/client/__init__.py

```diff
@@ -1582,7 +1582,7 @@ def get_cli_event_returns(
                                          timeout=timeout,
                                          tgt=tgt,
                                          tgt_type=tgt_type,
-                                         expect_minions=(verbose or show_timeout),
+                                         expect_minions=(kwargs.pop('expect_minions', False) or verbose or show_timeout),
```

Comment:
IMHO, this could use some in-line documentation here so we can understand why it's being popped later on when we come back to it.

Reply:
sounds good.